docs/dev-notes/BACKGROUND-QUEUE.md

```diff
@@ -89,8 +87,50 @@ C. push metric opened for push notification Z
 
-See file `QueueQueryRunner` to learn more about how the queue gets the next task to execute and how it skips tasks by group. 
+See class `QueueQueryRunner` to learn more about how the queue gets the next task to execute and how it skips tasks by group. 
 
-Learn more by visiting files `QueueRunRequest.swift` and `QueueRequestManager.swift`.
+Learn more by visiting class `QueueRunRequest`.
+
+## What data is saved when a task is added to the background queue? 
+
+The queue is only responsible for performing the network requests *after* the SDK has changed it's internal state (example: save a profile ID to local device storage for future SDK calls).
+
+The queue runner tasks are all [small in logic where they just perform a HTTP request](https://github.com/customerio/customerio-android/blob/7e6b1d6724fd199b6a721e05a4726e48d6c19089/sdk/src/main/java/io/customer/sdk/queue/QueueRunner.kt#L39-L43). Not necessarily on purpose, but it just works out that way because you want to do all the local changes to the state of the SDK before running the task in the queue. 
+
+The queue acts as a sync where it sends *snapshots of events* that have happened in the SDK to the API. For example, let's say that you have an app that allows you to modify the first name of your profile. After the app user changes their first name in the app, your app wants to send the edited first name to your remote API. Your app should do the following events:
+1. Save the first name to local storage such as a database to act as a cache. This changes the local state of the app. 
+2. Add a task to the background queue to eventually send the new first name to the remote API. It is at this time that you *provide to the background queue the new first name value* to be sent to the remote API. It's like you are taking a snapshot of the state of the SDK at this time and giving it to the background queue. You do not want to query the app's storage for the new first name when the background queue runs. You want to provide the actual first name value to the queue when you add the task. 
+
+```swift
+let profileId = 5 // the identifier to query our app's local data storage for this profile
+let newFirstName = "Eddie"
+database.updateFirstName(id: profileId, firstName: newFirstName)
+
+// Do this:
+backgroundQueue.addTask(
+    "edit_profile",
+    EditProfileQueueTaskData(newFirstName)
+)
+
+// Do *not* do this:
+backgroundQueue.addTask(
+    "edit_profile",
+    EditProfileQueueTaskData(profileId) // when "edit_profile" task runs in the background queue, query the database for the first name for id 5
+)
+```
+
+Why this is important is to avoid a scenario like this:
+* Your first name in app is `Dana`. 
+* You change your first name to `Eddie`. Add adds task to background queue to update first name. 
+* Background queue task has not yet run yet. You change your first name to `Frankie`. Add task to background queue to update first name. 
```

Comment:
~Background queue task has not yet run yet.~ You change your first name **again** to `Frankie` **before the background queue runs**. Add **another** task to **the** background queue to update first name.